eventuals/lazy.h

```diff
@@ -11,19 +11,26 @@ namespace eventuals {
 
 ////////////////////////////////////////////////////////////////////////
 
-template <typename T, typename... Args>
-class _Lazy final {
+template <typename T_, typename... Args_>
+struct _Lazy final {
  public:
-  template <typename Tuple>
-  _Lazy(Tuple args)
-    : args_(std::move(args)) {}
+  template <typename... Args>
+  using Args = std::enable_if_t<sizeof...(Args_) == 0, _Lazy<T_, Args...>>;
```

Comment:
To ensure someone can't do `Lazy::Of<Foo>::Args<>::Args<int>`:

```suggestion
  template <typename Arg, typename... Args>
  using Args = std::enable_if_t<sizeof...(Args_) == 0, _Lazy<T_, Arg, Args...>>;
```